draft-ietf-mls-protocol.md

```diff
@@ -462,29 +462,30 @@ struct {
 } StructWithVectors;
 ~~~~~
 
-Such a vector can represent values with length from 0 bytes to 2^62 bytes.
+Such a vector can represent values with length from 0 bytes to 2^30 bytes.
 The variable-length integer encoding reserves the two most significant bits
 of the first byte to encode the base 2 logarithm of the integer encoding length
 in bytes.  The integer value is encoded on the remaining bits, in network byte
 order.  The encoded value MUST use the smallest number of bits required to
 represent the value.  When decoding, values using more bits than necessary MUST
 be treated as malformed.
 
-This means that integers are encoded on 1, 2, 4, or 8 bytes and can encode 6-,
-14-, 30-, or 62-bit values respectively.
+This means that integers are encoded on 1, 2, or 4 bytes and can encode 6-,
+14-, or 30-bit values respectively.
 
-| Prefix | Length | Usable Bits | Min        | Max                 |
-|:-------|:-------|:------------|:-----------|:--------------------|
-| 00     | 1      | 6           | 0          | 63                  |
-| 01     | 2      | 14          | 64         | 16383               |
-| 10     | 4      | 30          | 16384      | 1073741823          |
-| 11     | 8      | 62          | 1073741824 | 4611686018427387903 |
+| Prefix | Length  | Usable Bits | Min        | Max                 |
+|:-------|:--------|:------------|:-----------|:--------------------|
+| 00     | 1       | 6           | 0          | 63                  |
+| 01     | 2       | 14          | 64         | 16383               |
+| 10     | 4       | 30          | 16384      | 1073741823          |
+| 11     | invalid | -           | -          | -                   |
 {: #integer-summary title="Summary of Integer Encodings"}
 
-For example, the eight-byte sequence c2 19 7c 5e ff 14 e8 8c (in hexadecimal)
-decodes to the decimal value 151288809941952652; the four byte sequence 9d 7f 3e
-7d decodes to 494878333; the two byte sequence 7b bd decodes to 15293; and the
-single byte 25 decodes to 37 (as does the two byte sequence 40 25).
+Vectors that start with "11" prefix are invalid and MUST be rejected.
```

Comment:
An alternative approach here would be to leave the `11` prefix, but forbid encoded values larger than some limit.  That way, if there were some case where reusing this encoding would be useful (cTLS, maybe) and larger sizes were allowed, then it could just be reused.  It also just seems a bit ugly to encode range limitations in the formatting.

Suggest something like the following:


> The `11` prefix is included for completeness.  To avoid compatibility issues for devices with limited storage, however, vectors in MLS MUST NOT have a length greater than 2^32.  A vector with an encoded length greater than this limit MUST be rejected as malformed.